plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -569,20 +569,25 @@ private void configureWith(List<YamlSource> sources) throws ConfiguratorExceptio
             throw new ConfiguratorException("Invalid configuration: '"+path+"' isn't a valid path.");
         }
 
-        if (!Files.isDirectory(root)) {
-            return matcher.matches(root) ? Collections.singletonList(root) : Collections.emptyList();
-        }
-
-        try {
-            return Files.list(root)
-                    .filter(Files::isRegularFile) // only consider regular files, following symlinks
-                    .filter(matcher::matches)     // matching pattern
-                    .collect(toList());
+        try (Stream<Path> stream = Files.find(Paths.get(path), Integer.MAX_VALUE,
+                (next, attrs) -> !attrs.isDirectory() && !isHidden(next) && matcher.matches(next))) {
+            return stream.collect(toList());
         } catch (IOException e) {
             throw new IllegalStateException("failed config scan for " + path, e);
         }
     }
 
+    private static boolean isHidden(Path path) {
```

Comment:
Would it not be better to use `File.isHidden`?

Reply:
E.g：
/var/jenkins_home/casc_configs/..data/abc.yaml 

It is not needed， but File.isHidden can't handle

Reply:
Fair enough 😄

Reply:
Is it possible to merge into the master? 😀